mlblocks/mlpipeline.py

```diff
@@ -777,7 +782,10 @@ def fit(self, X=None, y=None, output_=None, start_=None, debug=False, **kwargs):
 
             self._fit_block(block, block_name, context, debug_info)
 
-            if (block_name != self._last_block_name) or (block_name in output_blocks):
+            if block_name == self._last_fit_block:
```

Comment:
I would move this check above line 776, so this is the first thing that we do on the loop.

The reason is that if we have a `start_` that falls *after* the `last_fit_block`, we will skip this and keep producing blocks until the end unnecessarily.

Reply:
fixed